jgrapht-core/src/test/java/org/jgrapht/GraphMetricsTest.java

```diff
@@ -234,4 +238,113 @@ public void testDirectedGraphs()
             assertEquals(minCycle, GraphMetrics.getGirth(graph));
         }
     }
+
+    private static long naiveCountTriangles(Graph<Integer, DefaultEdge> graph){
```

Comment:
Testing of private methods, is usually performed by making them "package private" which means you simply remove the protection modifier from the method. See https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html

Guava even has a special annotation called `@VisibleForTesting` just for these use cases. See also https://stackoverflow.com/questions/6913325/annotation-to-make-a-private-method-public-only-for-test-classes .

Reply:
I've used reflection because now everything in `alg` can call `naiveCountTriangles`, but I see your point.

Reply:
I know. Both solution have pros and cons.